src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvThread.cs

```diff
@@ -302,7 +300,7 @@ private void OnPost()
                 wasWork = DoPostWork();
                 wasWork = DoPostCloseHandle() || wasWork;
                 loopsRemaining--;
-            } while (wasWork && loopsRemaining > 0);
+            } while (!_stopping && wasWork && loopsRemaining > 0);
```

Comment:
It's not like that takes much time. Also, the loop exits immediately after `_workAdding` is empty.